### PR TITLE
Handle etcd compacted error

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #17857 
Signed-off-by: Cai.Zhang <cai.zhang@zilliz.com>

### CODE REVIEW

File: internal/rootcoord/proxy_manager.go [hunk 1]
```diff
@@ -105,13 +107,22 @@ func (p *proxyManager) startWatchEtcd(ctx context.Context, eventCh clientv3.Watc
...
+			// TODO @xiaocai2333: watch proxy by session WatchService.
 		case event, ok := <-eventCh:
 			if !ok {
 				log.Warn("stop watching etcd loop due to closed etcd event channel")
 				return
 			}
 			if err := event.Err(); err != nil {
-				// TODO do we need to retry watch etcd when ErrCompacted, but the init session func may not be idempotent so skip
+				if err == v3rpc.ErrCompacted {
+					err2 := p.WatchProxy()
```
Comment (xiaofan-luan) [inline internal/rootcoord/proxy_manager.go:118]: we might miss some proxy down here.

For instance:
ProxyA, ProxyB up, we called AddSessionFunc.

THen etcd err compacted, we retry and add
Found only proxyA, then we trigger AddSessionFunc for ProxyA again.
But we missed the ProxyB down.

Comment (xiaocai2333) [reply internal/rootcoord/proxy_manager.go:119]: we must handle it in the init function.

Comment (xiaocai2333) [reply internal/rootcoord/proxy_manager.go:119]: done

**Comment by jennyli-z** [comment]
rerun ut

**Comment by xiaocai2333** [comment]
rerun ut

**Pull request merged**